openapi-extension/src/main/java/io/ballerina/openapi/extension/doc/OpenApiInfoUpdaterTask.java

```diff
@@ -172,11 +169,20 @@ private SeparatedNodeList<MappingFieldNode> getUpdatedFields(AnnotationNode anno
         MappingConstructorExpressionNode annotationValue = annotationValueOpt.get();
         SeparatedNodeList<MappingFieldNode> existingFields = annotationValue.fields();
         Token separator = NodeFactory.createToken(SyntaxKind.COMMA_TOKEN);
+        boolean openApiDefAvailable = false;
         for (MappingFieldNode field : existingFields) {
+            if (field instanceof SpecificFieldNode) {
+                String fieldName = ((SpecificFieldNode) field).fieldName().toString();
+                openApiDefAvailable = Constants.OPEN_API_DEFINITION_FIELD.equals(fieldName);
+            }
             fields.add(field);
             fields.add(separator);
         }
-        fields.add(createOpenApiDefinitionField(servicePath));
+        if (openApiDefAvailable) {
+            fields.remove(fields.size() - 1);
```

Comment:
What is this index? does it depends on the how user define the config?

Reply:
No, as you can see in line 179 we are adding a comma-token after every field. Hence, if we do not add another field we need to remove the last comma-token.